draft-ietf-ppm-dap.md

```diff
@@ -447,19 +447,22 @@ We start with some basic type definitions used in other messages.
 /* ASCII encoded URL. e.g., "https://example.com" */
 opaque Url<1..2^16-1>;
 
-Duration uint64; /* Number of seconds elapsed between two instants */
-
 Time uint64; /* seconds elapsed since start of UNIX epoch */
 
-/* An interval of time of length duration, where start is included and (start +
-duration) is excluded. */
+/* A nonce used to uniquely identify a report in the context of a DAP task. It
+includes the (possibly-rounded) timestamp recording when the report was
+generated, and a random 16-byte value. Nonces are totally ordered: first by
+time (numerically), then by rand (lexicographically). */
 struct {
-  Time start;
-  Duration duration;
-} Interval;
+  Time time;
+  uint8 rand[16];
+} Nonce;
 
-/* A nonce used to uniquely identify a report in the context of a DAP task. */
-Nonce uint8[16];
+/* A batch is defined by a range of nonces, from start (inclusive) to end (exclusive). */
+struct {
+  Nonce start;
+  Nonce end;
+} Batch;
```

Comment:
(Just to check that I understand:) A batch `batch1` is "valid" if:
1. (Total ordering) `batch1.start.time < batch1.end.time` or `batch1.start.time == batch1.end.time` and `batch1.start.nonce < batch1.end.nonce`
2. (Non-overlapping) There is no prior batch `batch0` that intersects with `batch1`, i.e., there is at least one nonce contained in both batches
3. (Boundary aligned) `batch1.start.time` and `batch1.start.end` are both equal to `0 (mod time_precision)`

Anything I'm missing?